File: Revisao/incremental4/Fatura.java

class Fatura {
    private float total;

    public Fatura() {
        this.total = 0;
    }

    public float getTotal() {
        return total;
    }

    public void adicionarTransacao(Transacao transacao) {
        total += transacao.getValor();
        System.out.println("Transação de R$" + transacao.getValor() + " adicionada à fatura.");
    }

    public void limparFatura() {
        total = 0;
        System.out.println("Fatura limpa.");
    }
}

File: Revisao/python/fatura.py

class Fatura():
    def __init__(self):
        self.total = 0
        self.transacoes = []
    
    def adicionar_transacao(self, transacao):
        self.transacoes.append(transacao)
        self.total = self.total + transacao.valor
        print(f"Transacao de r${transacao.valor} adicionada com sucesso.")
        
    def limpar_fatura(self):
        self.total = 0
        self.transacoes.clear()
        print("Fatura limpa com sucesso.")